store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/handler/GetElementsBetweenSetsHandler.java

```diff
@@ -20,36 +20,79 @@
 import uk.gov.gchq.gaffer.accumulostore.retriever.impl.AccumuloIDBetweenSetsRetriever;
-import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
 import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.id.DirectedType;
+import uk.gov.gchq.gaffer.data.element.id.EntityId;
+import uk.gov.gchq.gaffer.data.elementdefinition.view.View;
+import uk.gov.gchq.gaffer.operation.Operation;
 import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.graph.SeededGraphFilters.IncludeIncomingOutgoingType;
 import uk.gov.gchq.gaffer.store.Context;
 import uk.gov.gchq.gaffer.store.Store;
 import uk.gov.gchq.gaffer.store.StoreException;
-import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
-import uk.gov.gchq.gaffer.user.User;
+import uk.gov.gchq.gaffer.store.operation.handler.FieldDeclaration;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
 
-public class GetElementsBetweenSetsHandler implements OperationHandler<GetElementsBetweenSets, CloseableIterable<? extends Element>> {
+import java.util.Arrays;
 
-    @Override
-    public CloseableIterable<? extends Element> doOperation(final GetElementsBetweenSets operation,
-                                                            final Context context, final Store store)
-            throws OperationException {
-        return doOperation(operation, context.getUser(), (AccumuloStore) store);
-    }
+public class GetElementsBetweenSetsHandler implements OperationHandler<Iterable<? extends Element>> {
+
+    private static final String INCLUDE_INCOMING_OUTGOING = "includeIncomingOutgoing";
 
-    public CloseableIterable<? extends Element> doOperation(final GetElementsBetweenSets operation,
-                                                            final User user, final AccumuloStore store)
-            throws OperationException {
+    private static final String DIRECTED_TYPE = "directedType";
+
+    @Override
+    public Iterable<? extends Element> _doOperation(final Operation operation, final Context context, final Store store) throws OperationException {
         try {
-            final IteratorSettingFactory iteratorFactory = store.getKeyPackage().getIteratorFactory();
-            return new AccumuloIDBetweenSetsRetriever(store, operation, user, iteratorFactory.getElementPreAggregationFilterIteratorSetting(operation.getView(), store),
-                    iteratorFactory.getElementPostAggregationFilterIteratorSetting(operation.getView(), store),
-                    iteratorFactory.getEdgeEntityDirectionFilterIteratorSetting(operation),
-                    iteratorFactory.getQueryTimeAggregatorIteratorSetting(operation.getView(), store));
+            final AccumuloStore accumuloStore = (AccumuloStore) store;
+            final View view = (View) operation.get("view");
+            final DirectedType directedType = (DirectedType) operation.get(DIRECTED_TYPE);
```

Comment:
If optional, does the method you pass this value to accept `Null` ?. examine the old deleted Operation and see what the default if any was required.

Reply:
I believe it does but I will check. The `DirectedType` class has null checks for its static methods.

Reply:
The `ClassicIteratorSettingsFactory` and `ByteEntityIteratorSettingsFactory` call `DirectedType.isEither()` which checks for a `null` value. The `DirectedEdge` isn't given a default value in the [GetElementsBetweenSets](https://github.com/gchq/Gaffer/blob/develop/store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/impl/GetElementsBetweenSets.java) class.